platform/sponge/src/main/java/me/lokka30/treasury/plugin/sponge/apiimpl/economy/UniqueAccountImpl.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.plugin.sponge.apiimpl.economy;
+
+import java.util.UUID;
+import me.lokka30.treasury.api.economy.EconomyProvider;
+import me.lokka30.treasury.api.economy.account.PlayerAccount;
+import org.spongepowered.api.service.economy.account.UniqueAccount;
+
+public class UniqueAccountImpl extends VirtualAccountImpl implements UniqueAccount {
```

Comment:
All Treasury-backed UniqueAccounts being VirtualAccounts seems to violate the docs of VirtualAccount:
> Represents an Account which is not tied to a particular User.
>
> Examples of virtual accounts include:
> * A global server account
> * A 'bank account', shared among multiple users
> * A account for a non-player Entity.

Should probably have VirtualAccountImpl become an `AbstractAccountImpl implements Account` and then extend it to VirtualAccountImpl/UniqueAccountImpl.

Reply:
The difference between a `UniqueAccount` and `VirtualAccount` is that `UniqueAccount` features a `UUID` as its identifier, nothing else. Hence, in order to remove some implementation burden we can safely extend the `VirtualAccount` implementation.

Reply:
I get that this is one fewer implementation to maintain, but it's still a violation of the docs. VirtualAccount is an empty interface extending Account specifically for non-player account types.

Reply:
.... ok I'll do an abstract account implementation